src/main/java/no/ntnu/mycbr/rest/controller/RetrievalController.java

```diff
@@ -1,239 +1,212 @@
 package no.ntnu.mycbr.rest.controller;
 
-import org.springframework.web.bind.annotation.RestController;
 import io.swagger.annotations.ApiOperation;
-import io.swagger.annotations.ApiResponse;
-import io.swagger.annotations.ApiResponses;
 import no.ntnu.mycbr.rest.Query;
+import no.ntnu.mycbr.rest.common.ApiResponseAnnotations.ApiResponsesDefault;
+import no.ntnu.mycbr.rest.service.SelfSimilarityRetrieval;
+
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
 import org.springframework.web.bind.annotation.*;
 import java.util.*;
 import org.json.simple.JSONArray;
 import org.json.simple.parser.JSONParser;
 import org.json.simple.parser.ParseException;
-import static no.ntnu.mycbr.rest.utils.RESTCBRUtils.getFullResult;
+
+import static no.ntnu.mycbr.rest.utils.QueryUtils.getFullResult;
+
+import static no.ntnu.mycbr.rest.common.ApiResponseAnnotations.*;
+import static no.ntnu.mycbr.rest.common.Constants.*;
 
 @RestController
 public class RetrievalController {
+
     private final Log logger = LogFactory.getLog(getClass());
 
     @ApiOperation(value = "getSimilarInstances", nickname = "getSimilarInstances")
-    @RequestMapping(method = RequestMethod.POST, path="/concepts/{conceptID}/casebases/{casebaseID}/retrieval", produces = "application/json")
-    @ApiResponses(value = {
-            @ApiResponse(code = 200, message = "Success", response = Query.class),
-            @ApiResponse(code = 401, message = "Unauthorized"),
-            @ApiResponse(code = 403, message = "Forbidden"),
-            @ApiResponse(code = 404, message = "Not Found"),
-            @ApiResponse(code = 500, message = "Failure")
-    })
+    @RequestMapping(method = RequestMethod.POST, path=PATH_CONCEPT_CB_ID+"/retrieval", produces=APPLICATION_JSON)
+    @ApiResponsesForQuery
     public Query getSimilarInstances(
-            @RequestParam(value="casebase", defaultValue="CaseBase0") String casebase,
-            @RequestParam(value="concept name", defaultValue="Car") String concept,
-            @RequestParam(value="amalgamation function", defaultValue="CarFunc") String amalFunc,
-            @RequestParam(required = false, value="no of returned cases",defaultValue = "-1") int k,
```

Comment:
Was *no of returned cases*. Now it is *k*. Does it change the API path? If so please list all API path changes.
Same about `value="concept name"`